### PR TITLE
feat: implement summary tag component

### LINKED ISSUE #17848: Implement Summary Tag component
Implement this component in the quo2 component library as per the design spec.

Reuse Quo2 components wherever applicable.

Designs
https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Design-System-for-Mobile?type=design&node-id=22665-205765&mode=design&t=snMU2691xY2S8qNY-4

Visual reference:

Acceptance Criteria:

component spec tests - https://github.com/status-im/status-mobile/blob/develop/doc/component-tests-overview.md
quo2 preview screen
export from quo2/core.cljs
following best practices - https://github.com/status-im/status-mobile/blob/develop/doc/new-guidelines.md

Note
Please align with design team before starting this

### PR DESCRIPTION
**Pull request by briansztamfater**
fixes #17848 

### Summary

Implements Summary tag component

https://github.com/status-im/status-mobile/assets/18485527/03da09dc-2c92-44cb-9692-05877910d949

#### Platforms

- Android
- iOS

### Steps to test

- Open Status
- Open quo preview
- Select tag > Summary tag
- Test component

status: ready

### CODE REVIEW

File: src/quo/components/tags/summary_tag/view.cljs [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+                              :token         (str (:value token-props) " " (:symbol token-props))
+                              :account       (:name account-props)
+                              :user          (:full-name user-props)
+                              :saved-address (:full-name saved-address-props)
+                              :collectible   (:name collectible-props)
+                              :network       (:network-name network-props)
+                              :address       address
+                              "")
+        customization-color (case type
+                              :account       (:customization-color account-props)
```
Comment (J-Son89) [inline src/quo/components/tags/summary_tag/view.cljs]: imo we should pass `customization-color` as it's own prop instead 👍 

i.e 
```
(defn- view-internal
  [{:keys [customization-color ...
           address type
           theme]
           ```

File: src/quo/components/tags/summary_tag/view.cljs [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+      :style  {:width  24
+               :height 24}}]
+    nil))
+
+(defn- view-internal
+  [{:keys [account-props user-props collectible-props network-props token-props saved-address-props
+           address type
+           theme]
+    :as   props}]
+  (let [text                (case type
```
Comment (J-Son89) [inline src/quo/components/tags/summary_tag/view.cljs]: I think we're making the api of this component really confusing here. Why not just have a "text" param and handle this outside in application code.  although I would probably change "text" to something like "label"

File: src/quo/components/tags/summary_tag/view.cljs [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+     {:full-name           (:full-name user-props)
+      :size                :xs
+      :profile-picture     (:profile-picture user-props)
+      :customization-color (:customization-color user-props)}]
+    :token
+    [rn/image
+     {:source (resources/get-token (keyword (string/lower-case (:symbol token-props))))
+      :style  {:width  24
+               :height 24}}]
+    nil))
```
Comment (J-Son89) [inline src/quo/components/tags/summary_tag/view.cljs:44]: I personally would prefer this as something like:
```
(defn- left-view
  [{:keys [type  emoji customization-color label image-source
           ]}]
  (case type
    :account
    [account-avatar/view
     {:customization-color customization-color
      :size                24
      :emoji               emoji
      :type                :default}]
    :network
    [rn/image
     {:source image-source
      :style  {:width  24
               :height 24}}]
    :saved-address
    [wallet-user-avatar/wallet-user-avatar
     {:full-name           label
      :size                :size-24
      :customization-color customization-color }]
    :collectible
    [rn/image
     {:source image-source
      :style  style/collectible-image}]
    :user
    [user-avatar/user-avatar
     {:full-name           label
      :size                :xs
      :profile-picture     image-source
      :customization-color customization-color}]
    :token
    [rn/image
     {:source image-source
      :style  {:width  24
               :height 24}}]
    nil))
 ```
 and also the `token` image resolving I prefer to do in the application code, as perhaps in some case they are urls and others based on images that we already have in the codebase.
 
 cc @ulisesmac

File: src/quo/components/tags/summary_tag/view.cljs [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    [account-avatar/view
+     {:customization-color (:customization-color account-props)
+      :size                24
+      :emoji               (:emoji account-props)
+      :type                :default}]
+    :network
+    [rn/image
+     {:source (:network-logo network-props)
+      :style  {:width  24
+               :height 24}}]
```
Comment (ulisesmac) [inline src/quo/components/tags/summary_tag/view.cljs]: This could be in the style ns :)

File: src/quo/components/tags/summary_tag/view.cljs (hunk 3 above)
Comment (ulisesmac) [reply src/quo/components/tags/summary_tag/view.cljs:44]: Yep, I agree. The API looks cleaner :+1:

File: src/quo/components/tags/summary_tag/view.cljs (hunk 2 above)
Comment (ulisesmac) [reply src/quo/components/tags/summary_tag/view.cljs]: I agree too, I think this component should only receive the params it needs to work. Right now it's hard to know what keys are expected in those maps (account-props, user-props, network-props, etc) to make this work.

Also a docstring specifying the props supported for each type could work :+1:

**Comment by briansztamfater** [comment]
@ulisesmac @J-Son89 thanks for the feedback, will refactor and simplify the component API :)

**Review by J-Son89 — APPROVED** [review]

File: src/quo/components/tags/summary_tag/component_spec.cljs [hunk 5]
```diff
@@ -0,0 +1,51 @@
+(ns quo.components.tags.summary-tag.component-spec
+  (:require
+    [quo.components.tags.summary-tag.view :as summary-tag]
+    [test-helpers.component :as h]))
+
+(h/describe "Summary tag component tests"
```
Comment (vkjr) [inline src/quo/components/tags/summary_tag/component_spec.cljs:6]: I'd suggest adding one more test where the component instantiated with empty props, to make sure it doesn't crash

Comment (vkjr) [reply src/quo/components/tags/summary_tag/component_spec.cljs:6]: Similarly, since we have a set of props packed under `:token-props`, `:account-props`, their empty state also worth testing.

❗  Right now component crashes the app in following scenarios:
```clojure
[quo/summary-tag
         {:type        :token
          :token-props {}}]
```
this also:
```clojure
[quo/summary-tag
         {:type        :account
          :account-props {}}]
```          
          
Didn't test the rest

**Review by vkjr — CHANGES_REQUESTED** [review]
@briansztamfater, thanks for the PR!
Could you please also add tests to cover the cases when empty `:token-props`, `:account-props`, etc... passed to the component and make sure the component doesn't crash? 
Thanks!

**Comment by briansztamfater** [comment]
> @briansztamfater, thanks for the PR! Could you please also add tests to cover the cases when empty `:token-props`, `:account-props`, etc... passed to the component and make sure the component doesn't crash? Thanks!

@vkjr Thanks for the feedback, I simplified the API so there is no token-props, account-props, etc anymore. Also refactored tests and added a case where we pass no props so we make sure it doesn't crash.

**Review by vkjr — APPROVED** [review]

**Comment by briansztamfater** [comment]
@Francesca-G can you please review this component? Thanks in advance!

**Review by Francesca-G — APPROVED** [review]
Looks good :)
Here's the [review](https://www.figma.com/file/Tf5nfkYvpbnNCo4rKLK7lS/Feedback-for-Mobile?type=design&node-id=3415%3A214763&mode=design&t=RdhOk7x2IfbSaoue-1) with some minor issues to check

**Comment by briansztamfater** [comment]
@Francesca-G thanks for the feedback, updated the text in preview for the token variant (it is just a bug in the preview) and checked the height that is actually 32 points



Proceeding to merge, thanks!
